src/app/(nextjs_migration)/layout.tsx

```diff
@@ -2,18 +2,21 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-import { ReactNode } from 'react'
-import Script from 'next/script'
+import { ReactNode } from "react";
+import Script from "next/script";
+import { L10nProvider } from "../../contextProviders/localization";
+import { getL10nBundles } from "../functions/server/l10n";
 
-export default async function MigrationLayout ({
-  children
+export default async function MigrationLayout({
+  children,
 }: {
-  children: ReactNode,
+  children: ReactNode;
 }) {
+  const l10nBundles = getL10nBundles();
   return (
-    <>
-      <Script src="/nextjs_migration/client/js/resizeObserver.js"/>
+    <L10nProvider bundleSources={l10nBundles}>
+      <Script src="/nextjs_migration/client/js/resizeObserver.js" />
       {children}
-    </>
-  )
+    </L10nProvider>
+  );
```

Comment:
The [Next.js docs say](https://nextjs.org/docs/getting-started/react-essentials#using-context-in-client-components):

> Note: You should render providers as deep as possible in the tree (...) This makes it easier for Next.js to optimize the static parts of your Server Components.

So on that note: could the client-side L10n provider possibly be added just to the layout inside `(authenticated)`? Since the guest layout might not need it?

Reply:
I think also need the L10n provider for the `(guest)` layout. The client-side `SignInButton` component need localization as well.